processor/k8sprocessor/kube/client.go

```diff
@@ -243,14 +295,51 @@ func (c *WatchClient) extractPodAttributes(pod *api_v1.Pod) map[string]string {
 	}
 
 	for _, r := range c.Rules.Labels {
-		if v, ok := pod.Labels[r.Key]; ok {
-			tags[r.Name] = c.extractField(v, r)
+		if r.From == "pod" {
+			if v, ok := pod.Labels[r.Key]; ok {
+				tags[r.Name] = c.extractField(v, r)
+			}
+		}
+	}
+
+	for _, r := range c.Rules.Annotations {
+		if r.From == "pod" {
+			if v, ok := pod.Annotations[r.Key]; ok {
+				tags[r.Name] = c.extractField(v, r)
+			}
+		}
+	}
+	return tags
+}
+
+func (c *WatchClient) extractNamespaceAttributes(namespace *api_v1.Namespace) map[string]string {
+	tags := map[string]string{}
+
+	if c.Rules.NamespaceUID {
+		uid := namespace.GetUID()
+		tags[tagNamespaceUID] = string(uid)
+	}
+
+	if c.Rules.NamespaceStartTime {
+		ts := namespace.GetCreationTimestamp()
+		if !ts.IsZero() {
+			tags[tagNamespaceStartTime] = ts.String()
+		}
```

Comment:
are `NamespaceUID` and `NamespaceStartTime` new rules we added or were they always supported? They don't seem to be very useful TBH unlike a Pod's start time. If they are new (we not present in previous release), I think we should drop them to protect users from unknowingly watching all namespaces in a cluster for seemingly little benefit.

Reply:
These were added as new metadata extracted from the namespace object. I agree they are not much useful and will drop them(Also we don't need them right now)